js/models/messages.js

```diff
@@ -2044,33 +2193,43 @@
         this.set({ endSessionType: 'ongoing' });
       }
 
-      if (message.isFriendRequest() && isSessionRequest) {
-        // Check if the contact is a member in one of our private groups:
-        const groupMember = window
-          .getConversations()
-          .models.filter(c => c.get('members'))
-          .reduce((acc, x) => window.Lodash.concat(acc, x.get('members')), [])
-          .includes(primarySource);
-
-        if (groupMember) {
-          window.log.info(
-            `Auto accepting a 'group' friend request for a known group member: ${primarySource}`
-          );
-
-          window.libloki.api.sendBackgroundMessage(message.get('source'));
+      if (message.isSessionRequest()) {
+        await this.handleSessionRequest(source, primarySource, confirm);
 
-          confirm();
-        }
-
-        // Wether or not we accepted the FR, we exit early so background friend requests
+        // Wether or not we accepted the FR, we exit early so session requests
         // cannot be used for establishing regular private conversations
         return null;
       }
 
+      // Session request have been dealt with before, so a friend request here is
+      // not a session request message. Also, handleAutoFriendRequestMessage() only handles the autoAccept logic of an auto friend request.
+      // FIXME audric: is that correct? Also, if you recover your account from mnemonic for instance,
```

Comment:
fixme is about that case:
`(!isGroupMessage && !conversationOrigin.isFriend())`

Reply:
Well actually in the case of recover from mnemonic, when you receive a message you won't be able to decrypt, thus the session restoration logic will be triggered asking you wether to restore the session. Accepting will restart the friend request.

If we got a friend request that was sent before the time we restored, we should ignore it as it will cause more problems than needed.

Reply:
I removed the comment. I guess the case is needed but I don't understand why

Reply:
Neither can i, it's probably a weird edge case that was fixed but no comment was left :(